liquid-rust-middle/src/ty/mod.rs

```diff
@@ -276,10 +278,17 @@ impl AdtDef {
         &self.0.sorts
     }
 
+    pub fn generics(&self) -> &Vec<ParamTy> {
```

Comment:
a `&Vec<T>` is rarely more useful than a `&[T]` so let's return the slice to hide implementation details.

```suggestion
    pub fn generics(&self) -> &[ParamTy] {
```